edu.cuny.hunter.streamrefactoring.core/src/edu/cuny/hunter/streamrefactoring/core/analysis/StreamStateMachine.java

```diff
@@ -1291,4 +1284,63 @@ private void propagateStreamInstanceProperty(Collection<InstanceKey> streamInsta
 		}
 		return ret;
 	}
+
+	/**
+	 * Prune call graph
+	 * 
+	 * @param cg:
+	 *            call graph
+	 * @return a pruned call graph
+	 */
+	public static Graph<CGNode> pruneCallGraph(final CallGraph cg) {
+		Graph<CGNode> partialGraph = GraphSlicer.prune(cg, new Predicate<CGNode>() {
+			@Override
+			public boolean test(CGNode node) {
+				return isStreamNode(node);
```

Comment:
How is this related to https://github.com/ponder-lab/Optimize-Java-8-Streams-Refactoring/issues/108#issue-274664134? Comments please.